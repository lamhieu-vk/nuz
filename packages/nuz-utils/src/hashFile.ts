import crypto from 'crypto'
import fs from 'fs'

function hashFile(path: string, hash: string): string {
  return crypto.createHash(hash).update(fs.readFileSync(path)).digest('hex')
}

export default hashFile
