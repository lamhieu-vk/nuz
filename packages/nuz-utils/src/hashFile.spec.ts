import * as func from './hashFile'
import hashFile from './hashFile'

describe('hashFile', () => {
  it('Exported as default', () => {
    expect(func.default).toBeDefined()
  })

  it('Should match to the snapshot', () => {
    const path = require.resolve('./hashFile')
    expect(hashFile(path, 'md5')).toMatchSnapshot()
  })
})
