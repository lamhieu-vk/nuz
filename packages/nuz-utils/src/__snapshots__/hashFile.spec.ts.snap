// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hashFile Should match to the snapshot 1`] = `"ad2fde891e0ae2f1aa2462e1fd1b5d88"`;
